examples/live-replay-to-igtv.example.ts

```diff
@@ -0,0 +1,129 @@
+    /**
+     * make sure you are streaming to the url
+     * the next step will send a notification / start your stream for everyone to see
+     */
+    const startInfo = await ig.live.start(broadcast_id);
+    // status should be 'ok'
+    console.log(startInfo);
+
+    /**
+     * now, your stream is running
+     * the next step is to get comments
+     * note: comments can only be requested roughly every 2s
+     */
+
+        // initial comment-timestamp = 0, get all comments
+    let lastCommentTs = await printComments(broadcast_id, 0);
+
+    // enable the comments
+    await ig.live.unmuteComment(broadcast_id);
+    /**
+     * wait 2 seconds until the next request.
+     * in the real world you'd use something like setInterval() instead of Bluebird.delay() / just to simulate a delay
+     */
+    // wait 2s
+    await Bluebird.delay(2000);
+    // now, we print the next comments
+    lastCommentTs = await printComments(broadcast_id, lastCommentTs);
+
+    // now we're commenting on our stream
+    await ig.live.comment(broadcast_id, 'A comment');
+
+    /**
+     * now, your stream is running, you entertain your followers, but you're tired and
+     * we're going to stop the stream
+     */
+    await ig.live.endBroadcast(broadcast_id);
+
+    // Get live thumbnails, required to post on IGTV
+    let data = await ig.live.getPostLiveThumbnails(broadcast_id)
+
+    // Download any thumb
+    let file = await new Promise((resolve) => https.get(data.thumbnails[0], (download) => {
+        let ds = [];
+        download.on("data", (d) => ds.push(d));
+        download.on("end", () => resolve(Buffer.concat(ds)))
+    }))
+
+    // (optional) Resize thumb to a vertical one
+    file = await sharp(file)
+        .resize(720, 1280)
+        .png()
+        .toBuffer()
+
+    // It will be a png, it must be converted to jpg
+    file = await pngToJpeg({quality: 100})(file)
+
+    // Upload the thumbnail with a broadcast id for a replay and get uploadId
+    let upload = await ig.upload.photo({file, broadcastId: broadcast_id})
+
+    let igtv = null
+    let currentRetry = 0
+    let maxRetry = 3
+    let retryDelay = 4
+    while (!igtv) {
+        // This endpoint can return an error "202 Accepted; Transcode not finished yet" if Instagram has not finished to process the previous upload, so retry later in this case
+        try {
+            igtv = await ig.media.configureToIgtv({
+                upload_id: upload.upload_id,
+                title: 'A title',
+                caption: 'A description',
+                igtv_share_preview_to_feed: '1',
+            })
+
+            console.log(`Live posted to IGTV : ${igtv.upload_id}`))
+        } catch (e) {
+            currentRetry++
+            if (currentRetry > maxRetry) {
+                throw e
+            } else {
+                await (new Promise(resolve => {
+                  setTimeout(resolve, currentRetry * retryDelay)
+                }))
+            }
+        }
+    }
```

Comment:
This could live in its own function and not pollute the current function. 

For example:

```ts
async function retryDelayed<T>(fn: () => Promise<T>, retryOptions: { retries: number, delayMs: number }): Promise<T> {
  let step = 0;
  while(step++ < retryOptions.retries) {
    try {
      return await fn();
    } catch(e) {
      if(step >= retryOptions.retries) throw e;
      
      await new Promise(r => setTimeout(r, step * retryOptions.delayMs));
    }
  }
}

//  used like this:
const user = await retryDelayed(() => ig.user.info(123), {retries: 2, delayMs: 2 * 1000 /* 2s */ });
```

Reply:
Thanks for the tips @Nerixyz .
But before doing this... I really can't understand, it's not working anymore. I made and tested this PR yesterday, and today, I got a 500 error with the endpoint "/api/v1/media/configure_to_igtv/".
I think that there is a problem with my picture upload because if I use in the configureToIgtv method an existing "upload_id" (that I get from the app directly, so i'm bypassing the "download, convert and upload thumb for broadcast" step), it works.
If you have any idea... You're welcome!

Reply:
content/site-policy/content-removal-policies/github-private-information-removal-policy.md

Reply:
> Thanks for the tips @Nerixyz .
> But before doing this... I really can't understand, it's not working anymore. I made and tested this PR yesterday, and today, I got a 500 error with the endpoint "/api/v1/media/configure_to_igtv/".
> I think that there is a problem with my picture upload because if I use in the configureToIgtv method an existing "upload_id" (that I get from the app directly, so i'm bypassing the "download, convert and upload thumb for broadcast" step), it works.
> If you have any idea... You're welcome!